### PR TITLE
Enable -Werror flag in developer build

### PR DESCRIPTION
**Pull request by staniewzki**
### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

File: cmake/helpers.cmake [hunk 1]
```diff
@@ -244,8 +244,9 @@ function(add_umf_target_compile_options name)
             target_compile_definitions(${name} PRIVATE -D_FORTIFY_SOURCE=2)
         endif()
         if(UMF_DEVELOPER_MODE)
-            target_compile_options(${name} PRIVATE -fno-omit-frame-pointer
-                                                   -fstack-protector-strong)
+            target_compile_options(
+                ${name} PRIVATE -fno-omit-frame-pointer
+                                -fstack-protector-strong -Werror)
```
Comment (bratpiorka) [inline cmake/helpers.cmake:249]: please add similar flag for windows and clang (check for any warnings left in CI)

**Comment by ldorau** [comment]
@staniewzki please test your PR on your private fork

**Comment by ldorau** [comment]
Please rebase again, there are conflicts.

**Review by lukaszstolarczuk — COMMENTED** [review]
👍

**Review by PatKamin — APPROVED** [review]

**Review by ldorau — APPROVED** [review]

**Pull request merged**